src/components/inputs/radio-input/radio-group.js

```diff
@@ -36,25 +52,49 @@ class Group extends React.PureComponent {
   }
 
   render() {
-    const DirectionWrapper =
-      this.props.direction === 'stack' ? Spacings.Stack : Spacings.Inline;
+    const optionElements = React.Children.map(
+      this.props.children,
+      (child, index) => {
+        // NOTE: Allowing to intersperse other elements than `Option`.
+        if (child && child.type.displayName === Option.displayName) {
+          return React.cloneElement(child, {
+            id: `${this.props.id}-${index}`,
+            name: this.props.name,
+            isChecked: this.props.value === child.props.value,
+            isDisabled: child.props.isDisabled || this.props.isDisabled,
+            isReadOnly: this.props.isReadOnly,
+            hasError: this.props.hasError,
+            hasWarning: this.props.hasWarning,
+            onChange: this.props.onChange,
+            onFocus: this.props.onFocus,
+            onBlur: this.props.onBlur,
+          });
+        }
+        return child;
+      }
+    );
+    if (this.props.direction === 'inline') {
+      return (
+        <div id={this.props.id}>
+          <Spacings.Inline
```

Comment:
Not here but maybe in the future. I like the notion of `components: {}` we recently adopted far more. So it would be `components: { spacing: <Spacings.Inline scale='' /> }` maybe. Is that worth of something to track as an issue?

Reply:
I see your point. In this case though, you really only have 2 choices: stack or inline. I'm not sure having the freedom of passing any component would make sense here 🤔

Reply:
Yeah, me neither fully. Just wanted to share the thought. One could argue (not saying has to) that having an consistent/expected API is also nice. You should go "How can I customise this?" "Ah, just like everywhere, maybe pass in the `components` thing.".